src/main/java/com/hubspot/jinjava/lib/filter/SelectAttrFilter.java

```diff
@@ -110,20 +110,32 @@ protected Object applyFilter(
       }
     }
 
+    String tempValue = generateTempVariable();
+    String expression = generateTempVariable(tempValue, attr);
     ForLoop loop = ObjectIterator.getLoop(var);
     while (loop.hasNext()) {
       Object val = loop.next();
+      interpreter.getContext().put(tempValue, val);
+
+      Object attrVal = interpreter.resolveELExpression(
+        expression,
+        interpreter.getLineNumber()
+      );
 
-      Object attrVal = new Variable(
-        interpreter,
-        String.format("%s.%s", "placeholder", attr)
-      )
-      .resolve(val);
       if (acceptObjects == expTest.evaluate(attrVal, interpreter, expArgs)) {
         result.add(val);
       }
     }
+    interpreter.getContext().removeResolvedExpression(expression);
 
     return result;
   }
+
+  private String generateTempVariable() {
+    return "jj_temp_" + Math.abs(ThreadLocalRandom.current().nextInt());
+  }
+
+  private String generateTempVariable(String tempValue, String expression) {
+    return String.format("%s.%s", tempValue, expression).trim();
```

Comment:
The `String.format` has some performance issues. So, for curiosity, I ran a [benchmark](https://gist.github.com/samukce/c12a520c2f5ca8db19f560a9572bc39f) to compare the approaches.

Those are the results:
```bash
Benchmark                                                Mode  Cnt   Score   Error  Units
StringConcatSimulation.concatWithConcatFunction            ss   25   1.900 ± 0.120  ms/op
StringConcatSimulation.concatWithPlus                      ss   25   2.079 ± 0.438  ms/op
StringConcatSimulation.concatWithStringBuilder             ss   25   1.871 ± 0.436  ms/op
StringConcatSimulation.concatWithStringBuilderWithCount    ss   25   1.720 ± 0.255  ms/op
StringConcatSimulation.concatWithStringFormat              ss   25  15.145 ± 2.836  ms/op
```

I would suggest using StringBuilder instead.
```suggestion
    return new StringBuilder()
                .append(tempValue)
                .append(".")
                .append(expression)
                .toString()
                .trim();
```